src/sql/parts/modelComponents/modelEditor/modelViewEditor.ts

```diff
@@ -44,31 +40,56 @@ export class ModelViewEditor extends BaseEditor {
 	public focus(): void {
 	}
 
-	public setInput(input: ModelViewInput, options?: EditorOptions): TPromise<void, any> {
+	async setInput(input: ModelViewInput, options?: EditorOptions): TPromise<void, any> {
 		if (this.input && this.input.matches(input)) {
 			return TPromise.as(undefined);
 		}
+
 		const parentElement = this.getContainer().getHTMLElement();
-		$(parentElement).clearChildren();
+		if (this.input instanceof ModelViewInput) {
+			if (this.input.container) {
+				if (this.input.options && this.input.options.retainContextWhenHidden) {
+					this.input.container.style.visibility = 'hidden';
+				} else {
+					parentElement.removeChild(this.input.container);
+				}
+			}
+		}
 
-		if (!this._modelViewMap.get(input.modelViewId)) {
-			let modelViewContainer = DOM.$('div.model-view-container');
-			let dialogPane = new DialogPane(input.title, input.modelViewId, () => undefined, this._instantiationService);
-			dialogPane.createBody(modelViewContainer);
-			this._modelViewMap.set(input.modelViewId, modelViewContainer);
+		if (!parentElement.contains(input.container)) {
+			parentElement.appendChild(input.container);
 		}
-		let element = this._modelViewMap.get(input.modelViewId);
-		DOM.append(parentElement, element);
+		input.container.style.visibility = 'visible';
+
+		await super.setInput(input, options);
+		this.doUpdateContainer();
+	}
+
+	private doUpdateContainer() {
+		const modelViewContainer = this.input && (this.input as ModelViewInput).container;
+		if (modelViewContainer) {
+			const frameRect = this._editorFrame.getBoundingClientRect();
+			const containerRect = modelViewContainer.parentElement.getBoundingClientRect();
 
-		return super.setInput(input, options);
+			modelViewContainer.style.position = 'absolute';
+			modelViewContainer.style.top = `${frameRect.top}px`;
+			modelViewContainer.style.left = `${frameRect.left - containerRect.left}px`;
+			modelViewContainer.style.width = `${frameRect.width}px`;
+			modelViewContainer.style.height = `${frameRect.height}px`;
+		}
 	}
 
 	/**
 	 * Updates the internal variable keeping track of the editor's size, and re-calculates the sash position.
 	 * To be called when the container of this editor changes size.
 	 */
 	public layout(dimension: Dimension): void {
-
+		if (this.input instanceof ModelViewInput) {
```

Comment:
What happens if I have one input open and then call layout and then switch inputs? Won't only the current input get updated?

Reply:
Yes. We only want the current input got updated

---
In reply to: [190708100](https://github.com/Microsoft/sqlopsstudio/pull/1483#discussion_r190708100) [](ancestors = 190708100)